### PR TITLE
Update changelog with repeater name data migration

### PR DESCRIPTION
**Pull request by SmittieC**
[Ticket](https://dimagi-dev.atlassian.net/jira/software/c/projects/SC/boards/196?modal=detail&selectedIssue=SC-2361)
[Data migration change](https://github.com/dimagi/commcare-hq/pull/32359)
Please note that the data migration can be optionally run and that a future change will run the data migration with a "normal" migration.

##### Environments Affected

All or some. Depending on the amount of Repeater records in the environment

##### Announce New Release

- [x] After merging, I will follow [these instructions](https://confluence.dimagi.com/display/saas/Announcing+changes+affecting+third+parties#Announcingchangesaffectingthirdparties-announcing)
to announce a new commcare-cloud release.

### CODE REVIEW

File: changelog/0065-Populate_repeater_names.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+context: |
+  Populate all repeaters' `name` fields with its respective connection settings' name.
+
+  This is optional, but is recommended to do for all environments having more than 200 repeaters. A migration will
+  be added in the future which will automatically apply these changes during a code deploy if they have not yet
+  been applied. If there are many repeaters, it will slow down the migration process.
+
+details: |
+  To prepare for a future change where we allow repeater names to be configurable, we added a `name` field to the
+  `repeaters_repeater` table. In the future change, this field will be popualted with the repeater's corresponding
```
Comment (millerdev) [inline changelog/0065-Populate_repeater_names.yml:18]:
```suggestion
  `repeaters_repeater` table. In the future change, this field will be populated with the repeater's corresponding
```

File: changelog/0065-Populate_repeater_names.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  
+  Go into the django shell
+  ```shell
+  $ commcare-cloud <env> django-manage shell
+  ```
+
+  Inside the django shell, run
+  ```python
+  $ from corehq.motech.repeaters.models import SQLRepeater
+  $ SQLRepeater.objects.count()
```
Comment (millerdev) [inline changelog/0065-Populate_repeater_names.yml:34]: nit: the prompt for python shells is not usually `$`, and the prompt character doesn't seem necessary here anyway.
```suggestion
  from corehq.motech.repeaters.models import SQLRepeater
  SQLRepeater.objects.count()
```

File: changelog/0065-Populate_repeater_names.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  connection settings' name by means of a migration. If there are many repeaters in the environment, this migration
+  might take a long time to complete, so it is recommended to apply the changes before that migration runs in order
+  speed up that migration's execution time.
+
+update_steps: |
+  To see how many repeaters there are in an environment, follow the steps below:
+  
+  Go into the django shell
+  ```shell
+  $ commcare-cloud <env> django-manage shell
```
Comment (millerdev) [inline changelog/0065-Populate_repeater_names.yml:28]: The prompt character doesn't seem necessary here, and makes it a little harder to copy/paste
```suggestion
  commcare-cloud <env> django-manage shell
```

File: changelog/0065-Populate_repeater_names.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  Inside the django shell, run
+  ```python
+  $ from corehq.motech.repeaters.models import SQLRepeater
+  $ SQLRepeater.objects.count()
+  ```
+
+  To run the data migration, execute the following command:
+  
+  ```shell
+  $ commcare-cloud <env> django-manage populate_repeater_names
```
Comment (millerdev) [inline changelog/0065-Populate_repeater_names.yml:40]:
```suggestion
  commcare-cloud <env> django-manage populate_repeater_names 
```

File: changelog/0065-Populate_repeater_names.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  To run the data migration, execute the following command:
+  
+  ```shell
+  $ commcare-cloud <env> django-manage populate_repeater_names 
+  ```
+  
+  Precautions:
+  * No downtime is needed for this operation however it is best to run the command during a period
+    of low traffic.
```
Comment (millerdev) [inline changelog/0065-Populate_repeater_names.yml:45]:
```suggestion
  * No downtime is needed for this operation. However, it is best to run the command during a period
    of low traffic.
```

**Review by millerdev — COMMENTED** [review]
Left a few nit-picky comments and a typo correction. Otherwise looks good.

File: changelog/0065-Populate_repeater_names.yml (hunk 5 above)
Comment (SmittieC) [reply changelog/0065-Populate_repeater_names.yml]: @millerdev thanks for all the checks / improvements. It is addressed :+1:

**Review by snopoke — COMMENTED** [review]
Some minor suggestions to wording etc (I thought I'd submitted this yesterday)

File: changelog/0065-Populate_repeater_names.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+date: 2022-12-13
+optional_per_env: yes
+# (optional) Min version of HQ that MUST be deployed before this change can be rolled out (commit hash)
+min_commcare_version: ec13eaccc55d14b708c4e25349edcfd6936fb4d1
+# (optional) Max version of HQ that can be deployed before this change MUST be rolled out (commit hash)
+max_commcare_version:
+context: |
+  Populate all repeaters' `name` fields with its respective connection settings' name.
+
+  This is optional, but is recommended to do for all environments having more than 200 repeaters. A migration will
```
[the remark this answers is not in the document]
Comment (SmittieC) [reply changelog/0065-Populate_repeater_names.yml:12]: Do you think we should include the number of repeaters/data forwarders as well then, or just let them run in in all env's where there are repeaters?

File: changelog/0065-Populate_repeater_names.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  be added in the future which will automatically apply these changes during a code deploy if they have not yet
+  been applied. If there are many repeaters, it will slow down the migration process.
+
+details: |
+  To prepare for a future change where we allow repeater names to be configurable, we added a `name` field to the
+  `repeaters_repeater` table. In the future change, this field will be popualted with the repeater's corresponding
+  connection settings' name by means of a migration. If there are many repeaters in the environment, this migration
+  might take a long time to complete, so it is recommended to apply the changes before that migration runs in order
+  speed up that migration's execution time.
```
[the remark this answers is not in the document]
Comment (SmittieC) [reply changelog/0065-Populate_repeater_names.yml]: Splendid rewording! Thanks!

**Review by snopoke — APPROVED** [review]

**Pull request merged**